packages/grafana-ui/src/components/Table/TableNG/TableNG.tsx

```diff
@@ -135,295 +181,364 @@ export function TableNG(props: TableNGProps) {
+  ]);
 
-  // First pass: Assign minimum widths to columns that need it
-  columns.forEach((column) => {
-    if (!column.width && column.minWidth! > sharedWidth) {
-      column.width = column.minWidth;
-      availableWidth -= column.width!;
-      fieldCountWithoutWidth -= 1;
-    }
-  });
+  const hasSubTable = false;
 
-  // Recalculate shared width after assigning minimum widths
-  sharedWidth = availableWidth / fieldCountWithoutWidth;
+  // todo: don't re-init this on each memoizedData change, only schema/config changes
+  const rowHeight = useRowHeight(columns, data, hasSubTable);
 
-  // Second pass: Assign shared width to remaining columns
-  columns.forEach((column) => {
-    if (!column.width) {
-      column.width = sharedWidth;
-    }
-    column.minWidth = COLUMN.MIN_WIDTH; // Ensure min-width is always set
-  });
+  // we need to have variables with these exact names for the localization to work properly
+  const itemsRangeStart = pageRangeStart;
+  const displayedEnd = pageRangeEnd;
 
-  return columns;
+  return (
+    <>
+      <DataGrid<TableRow, TableSummaryRow>
+        className={styles.grid}
+        ref={gridHandle}
+        columns={columns}
+        rows={renderedRows}
+        defaultColumnOptions={{
+          minWidth: 50,
+          resizable: true,
+          sortable: true,
+          // draggable: true,
+        }}
+        onColumnWidthsChange={(widths) => {
+          for (const [key, entry] of widths) {
+            onColumnResize?.(key, entry.width);
+          }
+        }}
+        sortColumns={sortColumns}
+        rowHeight={rowHeight}
+        headerRowClass={styles.dataGridHeaderRow}
+        headerRowHeight={noHeader ? 0 : undefined}
+        bottomSummaryRows={hasFooter ? [{}] : undefined}
+        renderers={{
+          renderRow: (key, rowProps) =>
+            renderRow(key, rowProps, [], panelContext, data, enableSharedCrosshair ?? false),
+        }}
+      />
+      {
+        enablePagination && (
+          <div className={styles.paginationContainer} ref={paginationWrapperRef}>
+            <Pagination
+              className="table-ng-pagination"
+              currentPage={page + 1}
+              numberOfPages={numPages}
+              showSmallVersion={smallPagination}
+              onNavigate={(toPage) => {
+                setPage(toPage - 1);
+              }}
+            />
+            {!smallPagination && (
+              <div className={styles.paginationSummary}>
+                {/* TODO: once old table is deprecated, we can update the localiziation
+                    string with the more consistent variable names */}
+                <Trans i18nKey="grafana-ui.table.pagination-summary">
+                  {{ itemsRangeStart }} - {{ displayedEnd }} of {{ numRows }} rows
+                </Trans>
+              </div>
+            )}
+          </div>
+        )
+      }
+    </>
+  );
 }
 
-export function myRowRenderer(
+// export function mapFrameToDataGrid({
```

Comment:
`mapFrameToDataGrid` is currently unused, but it is exported... since this is behind a feature flag, it's ok that we try to not bring this all the way back in at this point, right? I assume we will reference and repurpose a ton of stuff in here, but for now I just want to make sure we're starting from as blank a slate as we can.